docs/MEMORY-SETTINGS.md

```diff
@@ -0,0 +1,495 @@
+app instance can be started. If the app instance can be started, the memory
+overhead is added to the RAM value and the resulting value is used to
+set the memory *limits for the cgroup* of the app instance.
+
+#### VMM Overhead calculation
+
+As mentioned above, the VMM overhead is calculated automatically in runtime if
+the explicit value is not set. Currently, the VMM overhead is calculated as
+follows:
+
+* Base Overhead: 370 Mb
+* Page Tables Overhead: RAM * 0.025 (2.5% of RAM)
+* VCPU Overhead: 3 Mb per CPU assigned to a VM
+* Device MMIO Overhead: it fluctuates between 0.66 and 0.81 % of MMIO total size
+
+These values are from our estimations and can be changed in the future.
+
+#### Overriding the VMM overhead
+
+In most cases, the VMM overhead is calculated automatically and there is no need
+to set it explicitly. But there is a possibility to set the memory overhead
+manually.
+
+There are two settings that can be used to override the VMM overhead estimation:
+
+1. `memory.vmm.limit.MiB` is a setting in a global device configuration
+   that sets the VMM overhead for all the app instances. If this setting is set,
+   the VMM overhead is calculated as the value of this setting.
+2. `VMMMaxMem` is a setting in the app configuration that sets the VMM overhead
+   for the specific app instance. It is available in the app configuration on
+   the controller.
+
+The global `memory.vmm.limit.MiB` has priority over the app-specific `VMMMaxMem`
+setting. So if both are set the real value comes from the global
+`memory.vmm.limit.MiB` setting. If none of them is set, we count the
+overhead in runtime.
+
+## Recommendations
+
+Below are recommendations on how to use the memory settings for different
+components in EVE OS. For each setting, we provide guidelines on when it makes
+sense to increase or decrease the values, as well as the potential impact of
+setting them too high or too low.
+
+### Note on soft and hard limits
+
+Currently, the soft and hard limits are set to the same value. Theoretically,
+the soft limit should be set to a value that is lower than the hard limit, and
+it can be used to handle the memory pressure in the system more effectively.
+However, in practice, we have not tested this configuration yet and do not
+have precise recommendations on how to set the soft limit.
+
+### dom0_mem
+
+The `dom0_mem` setting controls the memory allocated to the EVE OS's system
+services: the `eve` cgroup. This cgroup includes the `eve/services` cgroup and
+the `eve/containerd` cgroup.
+
+#### When to increase `dom0_mem`?
+
+Increase `dom0_mem` if you observe that the overall memory usage of the EVE
+services and the containerd service is frequently reaching their memory limits.
+It can be identified by instances where the Out-Of-Memory (OOM) killer is fired
+with constraint `CONSTRAINT_MEMCG` and the `oom_memcg` value set `eve` cgroup:
+
+```text
+oom-kill:constraint=CONSTRAINT_MEMCG,...,oom_memcg=/eve/,...
```

Comment:
Maybe you already describe this somewhere, but where can the user see this oom-kill message, in dmesg and also in device logs?

Reply:
Yep, dmesg and the logs on the controller (device index). Will add it.